raspberrypi-fast-ts.c

```diff
@@ -376,7 +376,7 @@ static int fifo_read_ul(struct fifo_data *data, bool *has_value_out, unsigned lo
                 ok = errno;
                 perror("Couldn't convert received FIFO value to a number. strtoul");
                 goto fail_reset_buffer;
-            } else if ((end == '\0') || (end == data->buffer)) {
+            } else if ((end == "\0") || (end == data->buffer)) {
```

Comment:
Thanks for reporting! That's a bug, you're right, but the fix doesn't work I'm afraid. You can't compare strings with `==` in C

There's actually two things wrong here:
- we want to check the character `end` points to instead
- `end` points to the first character after the last converted digit in the string. If it's something other than the end of string (== null byte), that means there were non-digits in the string, i.e. an invalid value was received. So we should actually check that `*end !='\0'` instead

```suggestion
            } else if ((*end != '\0') || (end == data->buffer)) {
```